### ISSUE TITLE
Middleware in client side: context.app is undefined (1.0.0 rc2)

### ISSUE THREAD

**Issue by mgesmundo** [nuxt/nuxt.js#1181]
Hi,
using a simple middleware:
```js
export default function (context) {
  if (context.isClient) console.log(context.app)
}
```
the <code>app</code> value is now <code>undefined</code> when the middleware is executed on the client side.
Regards.

**Comment by awronski** [comment]
Similar error in asyncData 
https://github.com/nuxt/nuxt.js/issues/1178

**Comment by Atinux** [comment]
Fix with `1.0.0-rc3`, see https://github.com/nuxt/nuxt.js/commit/28d0e726416319fb8ed522c3ce80f124b74c2622

**Comment by mgesmundo** [comment]
Thanks @Atinux !
